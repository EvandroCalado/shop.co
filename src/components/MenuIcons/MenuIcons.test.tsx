import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MenuIcons } from '.';

describe('<MenuIcons />', () => {
  it('should render with default values', () => {
    render(<MenuIcons />);

    expect(screen.getByTestId('shop')).toBeInTheDocument();
    expect(screen.getByText(/2/i)).toBeInTheDocument();
    expect(screen.getByTestId('user')).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = render(<MenuIcons />);

    expect(container).toMatchSnapshot();
  });
});
