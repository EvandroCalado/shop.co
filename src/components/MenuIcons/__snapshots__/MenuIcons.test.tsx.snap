// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcons /> should render a snapshot 1`] = `
<div>
  <div
    class="flex items-center gap-4"
  >
    <div
      class="relative cursor-pointer rounded-full p-2 duration-150 hover:bg-[#f0f0f0]"
    >
      <svg
        class="lucide lucide-shopping-cart "
        data-testid="shop"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="21"
          r="1"
        />
        <circle
          cx="19"
          cy="21"
          r="1"
        />
        <path
          d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
        />
      </svg>
      <span
        class="absolute right-0 top-0 flex h-4 w-4 items-center justify-center rounded-full border border-white bg-black/95 text-[10px] text-white"
      >
        2
      </span>
    </div>
    <div
      class="cursor-pointer rounded-full p-2 duration-150 hover:bg-[#f0f0f0]"
    >
      <svg
        class="lucide lucide-circle-user "
        data-testid="user"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <circle
          cx="12"
          cy="10"
          r="3"
        />
        <path
          d="M7 20.662V19a2 2 0 0 1 2-2h6a2 2 0 0 1 2 2v1.662"
        />
      </svg>
    </div>
  </div>
</div>
`;
